src/server/bnetserver/REST/LoginRESTService.cpp

```diff
@@ -247,52 +247,103 @@ int32 LoginRESTService::HandlePost(soap* soapClient)
+                lastPlayedCharacter.CharacterName = fields[4].GetString();
+                lastPlayedCharacter.CharacterGUID = fields[5].GetUInt64();
+                lastPlayedCharacter.LastPlayedTime = fields[6].GetUInt32();
+            }
 
-        stmt = LoginDatabase.GetPreparedStatement(LOGIN_SEL_BNET_LAST_PLAYER_CHARACTERS);
-        stmt->setUInt32(0, accountInfo->Id);
-        if (PreparedQueryResult lastPlayerCharactersResult = LoginDatabase.Query(stmt))
-        {
-            Field* fields = lastPlayerCharactersResult->Fetch();
-            Battlenet::RealmHandle realmId{ fields[1].GetUInt8(), fields[2].GetUInt8(), fields[3].GetUInt32() };
-            Battlenet::Session::LastPlayedCharacterInfo& lastPlayedCharacter = accountInfo->GameAccounts[fields[0].GetUInt32()]
-                .LastPlayedCharacters[realmId.GetSubRegionAddress()];
-
-            lastPlayedCharacter.RealmId = realmId;
-            lastPlayedCharacter.CharacterName = fields[4].GetString();
-            lastPlayedCharacter.CharacterGUID = fields[5].GetUInt64();
-            lastPlayedCharacter.LastPlayedTime = fields[6].GetUInt32();
+            BigNumber ticket;
+            ticket.SetRand(20 * 8);
+
+            loginResult.set_login_ticket("TC-" + ByteArrayToHexStr(ticket.AsByteArray(20).get(), 20));
+
+            AddLoginTicket(loginResult.login_ticket(), std::move(accountInfo));
         }
+        else if (!accountInfo->IsBanned)
+        {
+            uint32 const maxWrongPassword = uint32(sConfigMgr->GetIntDefault("WrongPass.MaxCount", 0));
+
+            if (sConfigMgr->GetBoolDefault("WrongPass.Logging", false))
+                TC_LOG_DEBUG("server.rest", "[%s, Account %s, Id %u] Attempted to connect with wrong password!", ip_address.c_str(), login.c_str(), accountInfo->Id);
 
-        BigNumber ticket;
-        ticket.SetRand(20 * 8);
+            stmt = LoginDatabase.GetPreparedStatement(LOGIN_UPD_BNET_FAILED_LOGINS);
+            stmt->setUInt32(0, accountInfo->Id);
+            LoginDatabase.Execute(stmt);
 
-        loginResult.set_login_ticket("TC-" + ByteArrayToHexStr(ticket.AsByteArray(20).get(), 20));
+            accountInfo->FailedLogins++;
 
-        AddLoginTicket(loginResult.login_ticket(), std::move(accountInfo));
+            if (maxWrongPassword && accountInfo->FailedLogins >= maxWrongPassword)
+            {
+                BanMode const banType = BanMode(sConfigMgr->GetIntDefault("WrongPass.BanType", uint16(BanMode::BAN_IP)));
+                int32 const banTime = sConfigMgr->GetIntDefault("WrongPass.BanTime", 600);
+
+                if (banType == BanMode::BAN_ACCOUNT)
+                    BanBnetAccount(accountInfo->Id, banTime);
+                else
+                    BanIp(ip_address, banTime);
+
+                stmt = LoginDatabase.GetPreparedStatement(LOGIN_UPD_BNET_RESET_FAILED_LOGINS);
+                stmt->setUInt32(0, accountInfo->Id);
+                LoginDatabase.Execute(stmt);
```

Comment:
Just a question: Are PreparedStatement given to Transaction deleted during the transaction's commit?